### PR TITLE
fix table filter actions from accidentally sorting the column

### LINKED ISSUE #69: Filter actions have unintended behavior of sorting the column if sorting is enabled
#### Environment
- __Package version(s)__: 1.4.1
- __Operating System__: windows
- __Browser name and version__: Chrome Version 90.0.4430.212

#### Code Sandbox
Link to a minimal repro (fork [this code sandbox](https://codesandbox.io/s/itwinui-react-example-son74)): https://codesandbox.io/s/itwinui-react-example-forked-2sscm

#### Steps to reproduce
1. Go to the code sandbox above
2. Click filter icon on Name column
3. Observe the Sort Icon for the column and the rows (column is sorted)

1. Click filter icon on Name column
2. Type some string, press filter
3. Observe the Sort Icon for the column and the rows
4. Click filter icon again
5. Clear the filter
6.  Observe the Sort Icon for the column and the rows

#### Actual behavior
Column sorting changes

#### Expected behavior
Column should not be sorted while clicking filter icon, or on filter actions filter or clear

#### Additional information
Add anything else you want to mention (additional context, possible solution, etc.).

### PR DESCRIPTION
**Pull request by basanta078**
Thank you for your contribution to the iTwinUi-react project! Please make sure to complete all of the items below before submitting the PR.

## Checklist

- [x] Add meaningful tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] Add component features demo in Storybook (different stories)
- [x] Add an entry to the changelog for any new components and changes
- [x] Add screenshots of the key elements of the component below
- [x] Add any additional comments below describing the features you've implemented

## Screenshots
link to codebox: https://codesandbox.io/s/itwinui-react-example-forked-2sscm
## Additional Comments
closes #69

### CODE REVIEW

File: CHANGELOG.md
Comment (mayank99) [inline CHANGELOG.md:15]: Could reword to clarify.

```suggestion
- **Fixed `Table` bug where filter actions would also trigger sorting.**
```

File: src/core/Table/filters/FilterButtonBar.tsx
Comment (mayank99) [inline src/core/Table/filters/FilterButtonBar.tsx:66]: Really curious why clicking on a filter button (which is located in a dropdown, nowhere close to the actual table cell) would trigger sorting

Comment (mayank99) [reply src/core/Table/filters/FilterButtonBar.tsx:66]: @basanta078 Could you investigate the root cause of this?

File: CHANGELOG.md
Comment (basanta078) [reply CHANGELOG.md:11]: removed extra spaces

File: src/core/Table/filters/FilterButtonBar.tsx
Comment (basanta078) [reply src/core/Table/filters/FilterButtonBar.tsx:66]: even though filter button might look like it should not affect it, it is rendered for the same column and share the parent which is the column header

in react events propagate this way, popover's parent is the column header

Comment (basanta078) [reply src/core/Table/filters/FilterButtonBar.tsx]: i moved the logic to base filter to prevent all clicks going to column header.
it sorted column even on clicking the text box
now it should fix all accidental clicks from the filter ui

**Comment by bentleyvk** [comment]
I have also made a fix in my branch 😀 So I don't know whether to wait until I merge it or fix this one.

**Comment by basanta078** [comment]
> I have also made a fix in my branch 😀 So I don't know whether to wait until I merge it or fix this one.

what is in your fix? I want these change asap

File: src/core/Table/filters/FilterButtonBar.test.tsx
Comment (mayank99) [inline src/core/Table/filters/FilterButtonBar.test.tsx:79]: This test is failing now.

Comment (basanta078) [reply src/core/Table/filters/FilterButtonBar.test.tsx:80]: fixed it, had to move the logic to basefilter as input box click was also toggling the sort

**Comment by bentleyvk** [comment]
Does it fully fix the issue? Because I had to do this:
https://github.com/iTwin/iTwinUI-react/pull/63/files#diff-b06d7a4cfaae61f1e80cef514edb253a0bc94f591a82b3962840bf04f022f598R52

**Comment by basanta078** [comment]
> Does it fully fix the issue? Because I had to do this:
> https://github.com/iTwin/iTwinUI-react/pull/63/files#diff-b06d7a4cfaae61f1e80cef514edb253a0bc94f591a82b3962840bf04f022f598R52

I ignored the pointer issue, it is only cosmetic. This PR fixes the accidental sorting issue as the clicks don't go to header anymore
As for the pointer on hover it needs to be fixed from header css itself, and is not really causing issue here for this defect

**Comment by basanta078** [comment]
> > Does it fully fix the issue? Because I had to do this:
> > https://github.com/iTwin/iTwinUI-react/pull/63/files#diff-b06d7a4cfaae61f1e80cef514edb253a0bc94f591a82b3962840bf04f022f598R52
> 
> I ignored the pointer issue, it is only cosmetic. This PR fixes the accidental sorting issue as the clicks don't go to header anymore
> As for the pointer on hover it needs to be fixed from header css itself, and is not really causing issue here for this defect

I think this PR is good for now, @bentleyvk @mayank99 let me know if you can review this and let me know

File: src/core/Table/filters/BaseFilter.test.tsx
Comment (bentleyvk) [inline src/core/Table/filters/BaseFilter.test.tsx:1]: It is the same as FilterButtonBar so I don't think it is worth having duplicate tests. Better test filter icon click as clicking it also triggered sorting before.

Comment (basanta078) [reply src/core/Table/filters/BaseFilter.test.tsx]: removed it, and I added your filter icon test

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**
